core/types/transaction.go

```diff
@@ -205,14 +200,17 @@ func (tx *Transaction) decodeTyped(b []byte) (TxData, error) {
 	if len(b) <= 1 {
 		return nil, errShortTypedTx
 	}
+
+	if inner, isCelo, err := celoDecodeTyped(b); isCelo {
```

Comment:
I'd say the overhead is better amortised when it's many things. But even with one thing not having modifications in the middle of the switch statement will likely reduce the chance of conflicts because all new transaction additions will modify the switch statement.